### ISSUE TITLE
Make our integration tests self-contained

### ISSUE THREAD

**Issue by thomaseizinger** [comit-network/comit-rs#134]
Currently, we have two different types of tests:
- Independent unit tests
- Integration tests that depend on running instances of ganache or bitcoind

This imposes the following problems on testing:
- We cannot run tests in parallel because ganache and bitcoind don't support this
- We cannot just do `cargo test --all` because the environment is not set up

Both of these problems can be solved by making the tests self-contained, e.g. every test sets up everything that is needed for the test to run. In order to achieve this, we would have to 

- start docker containers from inside the tests
- figure out their port (i.e. let docker choose one that is available)
- wait for the docker container to be ready
- create a new BitcoinClient or whatever we want and point it to the created docker instance
- do the test
- shut down the container

I am picturing something similiar as https://www.testcontainers.org/.
Basically a docker client for rust that has support for specific images so that we can implement some kind of "waiting strategy".
